src/sentry/api/endpoints/organization_member_details.py

```diff
@@ -147,7 +169,43 @@ def put(self, request, organization, member_id):
         if has_sso:
             sso_enabled.send(organization=organization, sender=request.user)
 
-        return Response(status=204)
+        with transaction.atomic():
+            if result.get('teams'):
+                # dupe code from member_index
+                # ensure listed teams are real teams
+                teams = list(Team.objects.filter(
```

Comment:
it would be nice to cover this branch with a test

Reply:
Added here: https://github.com/getsentry/sentry/pull/6445/commits/13b36d8132947c9c4a41377cf27686cb6efc298d